### ISSUE TITLE
Visdom python client failed to establish socket

### ISSUE THREAD

**Issue by cdepillabout** [junyanz/pytorch-CycleGAN-and-pix2pix#455]
I'm getting an error saying that Visdom client is not able to connect.

The following is how I am running `train.py` and the output:

```sh
$ python train.py --dataroot datasets/maps --name maps_cyclegan --model cycle_gan --gpu_ids -1 --loadSize 143 --fineSize 128 --verbose --load_iter 0
----------------- Options ---------------
               batch_size: 1                             
                    beta1: 0.5                           
          checkpoints_dir: ./checkpoints                 
           continue_train: False                         
                 dataroot: datasets/maps                        [default: None]
             dataset_mode: unaligned                     
                direction: AtoB                          
              display_env: main                          
             display_freq: 400                           
               display_id: 1                             
            display_ncols: 4                             
             display_port: 8097                          
           display_server: http://localhost              
          display_winsize: 256                           
                    epoch: latest                        
              epoch_count: 1                             
                 fineSize: 128                                  [default: 256]
                  gpu_ids: -1                                   [default: 0]
                init_gain: 0.02                          
                init_type: normal                        
                 input_nc: 3                             
                  isTrain: True                                 [default: None]
                 lambda_A: 10.0                          
                 lambda_B: 10.0                          
          lambda_identity: 0.5                           
                 loadSize: 143                                  [default: 286]
                load_iter: 0                                    [default: 0]
                       lr: 0.0002                        
           lr_decay_iters: 50                            
                lr_policy: lambda                        
         max_dataset_size: inf                           
                    model: cycle_gan                     
               n_layers_D: 3                             
                     name: maps_cyclegan                        [default: experiment_name]
                      ndf: 64                            
                     netD: basic                         
                     netG: resnet_9blocks                
                      ngf: 64                            
                    niter: 100                           
              niter_decay: 100                           
               no_dropout: True                          
                  no_flip: False                         
                  no_html: False                         
                 no_lsgan: False                         
                     norm: instance                      
              num_threads: 4                             
                output_nc: 3                             
                    phase: train                         
                pool_size: 50                            
               print_freq: 100                           
           resize_or_crop: resize_and_crop               
             save_by_iter: False                         
          save_epoch_freq: 5                             
         save_latest_freq: 5000                          
           serial_batches: False                         
                   suffix:                               
         update_html_freq: 1000                          
                  verbose: True                                 [default: False]
----------------- End -------------------
dataset [UnalignedDataset] was created
#training images = 1096
initialize network with normal
initialize network with normal
initialize network with normal
initialize network with normal
model [CycleGANModel] was created
---------- Networks initialized -------------
ResnetGenerator(
  (model): Sequential(
    (0): ReflectionPad2d((3, 3, 3, 3))
    (1): Conv2d(3, 64, kernel_size=(7, 7), stride=(1, 1))
    (2): InstanceNorm2d(64, eps=1e-05, momentum=0.1, affine=False, track_running_stats=False)
    (3): ReLU(inplace)
...
    (6): InstanceNorm2d(256, eps=1e-05, momentum=0.1, affine=False, track_running_stats=False)
    (7): LeakyReLU(negative_slope=0.2, inplace)
    (8): Conv2d(256, 512, kernel_size=(4, 4), stride=(1, 1), padding=(1, 1))
    (9): InstanceNorm2d(512, eps=1e-05, momentum=0.1, affine=False, track_running_stats=False)
    (10): LeakyReLU(negative_slope=0.2, inplace)
    (11): Conv2d(512, 1, kernel_size=(4, 4), stride=(1, 1), padding=(1, 1))
  )
)
[Network D_B] Total number of parameters : 2.765 M
-----------------------------------------------
'>' not supported between instances of 'float' and 'NoneType'
'>' not supported between instances of 'float' and 'NoneType'
'>' not supported between instances of 'float' and 'NoneType'
Visdom python client failed to establish socket to get messages from the server. This feature is optional and can be disabled by initializing Visdom with `use_incoming_socket=False`, which will prevent waiting for this request to timeout.
create web directory ./checkpoints/maps_cyclegan/web...
'>' not supported between instances of 'float' and 'NoneType'
'>' not supported between instances of 'float' and 'NoneType'
```

The Visdom error is near the end.  Here's the error in question:

```
Visdom python client failed to establish socket to get messages from the
server. This feature is optional and can be disabled by initializing Visdom with
`use_incoming_socket=False`, which will prevent waiting for this request to timeout.
```

It says this feature is optional, so maybe I don't need to worry about it.

Also, the file `./checkpoints/maps_cyclegan/web/index.html` is not created within the first few minutes of running the above command, but maybe it takes a while before it is created?  (Although the directory `./checkpoints/maps_cyclegan/web/` does exist.)

If I open up the Visdom server at http://localhost:8097, I just see the default blue screen, but I don't see anything that appears to be output from `train.py`.  Maybe it takes a while before something appears?

---------------------------------------------------------------------------------------------------

In case anyone is trying to reproduce this, I have a `nix-shell` script I've used to create an environment to run `train.py`:

https://github.com/cdepillabout/pytorch-CycleGAN-and-pix2pix/blob/3048038eab83ff4117698680f7a7b9273315a5cb/shell.nix

**Comment by cdepillabout** [comment]
Also, it looks like the `'>' not supported between instances of 'float' and 'NoneType'` error may have been introduced by the fix to the issue below?

https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/issues/440

**Comment by junyanz** [comment]
I cannot reproduce your bug. Are you using the latest code?

**Comment by cdepillabout** [comment]
@junyanz 

> Are you using the latest code?

I'm using the latest checkout from `master`.

I'm not sure why you're not able to reproduce this.  Maybe it has to do with different `python` versions?  I'm using Python 3.7.1.

> file ./checkpoints/maps_cyclegan/web/index.html is not created within the first few minutes of running the above command, but maybe it takes a while before it is created?

After running `train.py` for a while using the above command, I have confirmed that this `index.html` is created.

> If I open up the Visdom server at http://localhost:8097, I just see the default blue screen, but I don't see anything that appears to be output from train.py. Maybe it takes a while before something appears?

I have confirmed that this actually does work as well.  I was correct in my guess that it takes a while before something appears.

> `'>' not supported between instances of 'float' and 'NoneType'`

This error still repeatedly shows up.

------------------------------------------------------------------------

To sum up, I'm still seeing the error `Visdom python client failed to establish socket to get messages from the server.`.  I'm also seeing the error `'>' not supported between instances of 'float' and 'NoneType'`.

However, it doesn't appear that either of these are actually causing problems, as `train.py` appears to be working.

**Comment by taesung89** [comment]
This could be some version problem with visdom or its dependencies. Could you look at this post ([https://github.com/facebookresearch/visdom/issues/450](https://github.com/facebookresearch/visdom/issues/450))?

**Comment by cdepillabout** [comment]
@taesung89 Thanks for responding.

facebookresearch/visdom#450 does look like the problem I am seeing.

I guess this issue can be closed as a known problem with visdom?  It doesn't appear to affect anything that I can tell, so closing is probably fine.

**Comment by taesung89** [comment]
Sounds great! Thanks for quick response.
